examples/ecs/iter_combinations.rs

```diff
@@ -0,0 +1,135 @@
+    let vel_range = -0.5..0.5;
+
+    let mut rng = thread_rng();
+    for _ in 0..200 {
+        let mass_value_cube_root: f32 = rng.gen_range(0.9..8.0);
+        let mass_value: f32 = mass_value_cube_root * mass_value_cube_root * mass_value_cube_root;
+
+        commands.spawn_bundle(BodyBundle {
+            pbr: PbrBundle {
+                transform: Transform {
+                    translation: Vec3::new(
+                        rng.gen_range(-1.0..1.0),
+                        rng.gen_range(-1.0..1.0),
+                        rng.gen_range(-1.0..1.0),
+                    )
+                    .normalize()
+                        * rng.gen_range(pos_range.clone()),
+                    scale: Vec3::splat(mass_value_cube_root * 0.1),
+                    ..Default::default()
+                },
+                mesh: mesh.clone(),
+                material: materials.add(
+                    Color::rgb_linear(
+                        rng.gen_range(color_range.clone()),
+                        rng.gen_range(color_range.clone()),
+                        rng.gen_range(color_range.clone()),
+                    )
+                    .into(),
+                ),
+                ..Default::default()
+            },
+            mass: Mass(mass_value),
+            velocity: Velocity(Vec3::new(
+                rng.gen_range(vel_range.clone()),
+                rng.gen_range(vel_range.clone()),
+                rng.gen_range(vel_range.clone()),
+            )),
+        });
+    }
+
+    // add bigger "star" body in the center
+    commands
+        .spawn_bundle(BodyBundle {
+            pbr: PbrBundle {
+                transform: Transform {
+                    scale: Vec3::splat(3.0),
+                    ..Default::default()
+                },
+                mesh: meshes.add(Mesh::from(shape::Icosphere {
+                    radius: 1.0,
+                    subdivisions: 5,
+                })),
+                material: materials.add((Color::ORANGE_RED * 10.0).into()),
+                ..Default::default()
+            },
+            mass: Mass(1500.0),
+            velocity: Velocity(Vec3::ZERO),
+        })
+        .insert(Light {
+            color: Color::ORANGE_RED,
+            ..Default::default()
+        });
+    commands.spawn_bundle(PerspectiveCameraBundle {
+        transform: Transform::from_xyz(0.0, 10.5, -20.0).looking_at(Vec3::ZERO, Vec3::Y),
+        ..Default::default()
+    });
+}
+
+fn interact_bodies(mut query: Query<(&Mass, &GlobalTransform, &mut Velocity)>) {
+    let mut iter = query.iter_combinations_mut();
+    while let Some([(Mass(m1), transform1, mut vel1), (Mass(m2), transform2, mut vel2)]) =
+        iter.fetch_next()
+    {
+        let delta = transform2.translation - transform1.translation;
+        let distance_sq: f32 = delta.length_squared();
+        let delta_norm = delta / distance_sq.sqrt();
+        let force = delta_norm * (GRAVITY_CONSTANT * (m1 + m2) / distance_sq.max(0.01));
```

Comment:
I guess this should be product of masses here and not sum:
```suggestion
        let force = delta_norm * (GRAVITY_CONSTANT * m1 * m2 / distance_sq.max(0.01));
```

Reply:
Yeah, good catch. I've actually ended up rewriting the force calculation and integration step, as everything was wrong about it, including the timestep. Now it should resemble actual physics a little more 😄